src/hooks/vaultsHooks.ts

```diff
@@ -0,0 +1,426 @@
+            totalPnl: '0',
+            subaccountId: '0',
+            netTransfers: '0',
+            blockHeight: '0',
+            blockTime: '0',
+          },
+        ],
+      },
+    ],
+  });
+}
+
+async function placeholderFetchMegavaultPositions() {
+  await delay(Math.random() * 2000);
+  return JSON.stringify({
+    positions: [
+      {
+        ticker: 'BTC-USD',
+        assetPosition: {
+          side: 'LONG',
+          size: '100',
+          assetId: 'USDC',
+          subaccountNumber: 0,
+        },
+        perpetualPosition: {
+          market: 'BTC-USD',
+          status: 'OPEN',
+          side: 'SHORT',
+          size: '100',
+          maxSize: '100',
+          entryPrice: '40000',
+          realizedPnl: '0',
+          createdAt: '0',
+          createdAtHeight: '0',
+          sumOpen: undefined,
+          sumClose: undefined,
+          netFunding: '0',
+          unrealizedPnl: undefined,
+          closedAt: undefined,
+          exitPrice: undefined,
+          subaccountNumber: 0,
+        },
+        equity: '100',
+      },
+    ],
+  });
+}
+
+async function placeholderFetchVaultAccount() {
+  await delay(Math.random() * 2000);
+
+  const baseObj = {
+    address: '0x123',
+    shares: 100,
+    locked_shares: 0,
+    equity: 100,
+    withdrawable_amount: 100,
+  };
+  return JSON.stringify(baseObj);
+}
+
+async function placeholderFetchVaultFormSlippage(shares: number, resultAmountTemporary: number) {
+  await delay(Math.random() * 2000);
+
+  const baseObj = {
+    shares,
+    expectedAmount: resultAmountTemporary,
+  };
+  return JSON.stringify(baseObj);
+}
+
+async function placeholderFetchVaultAccountTransfers() {
+  await delay(Math.random() * 2000);
+
+  const baseObj = {
+    pageSize: 100,
+    totalResults: 0,
+    offset: 0,
+    transfersSubset: [],
+    totalNetTransfers: '0',
+  };
+  return JSON.stringify(baseObj);
+}
+
+function useDebounceHf<T>(value: T, delayMs?: number): T {
+  const [debouncedValue, setDebouncedValue] = useState<T>(value);
+
+  useEffect(() => {
+    const timer = setTimeout(() => setDebouncedValue(value), delayMs ?? 500);
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [value, delayMs]);
+
+  return debouncedValue;
+}
+
+// it's illegal to return undefined from use query so we just wrap results in a data object
+function wrapNullable<T>(data: T | undefined | null): { data: T | null | undefined } {
+  return { data };
+}
+
+const vaultQueryOptions = {
+  staleTime: 1000 * 60,
+  refetchInterval: 1000 * 60 * 2,
+};
+
+export const useLoadedVaultDetails = () => {
+  const { data: vaultDetails } = useQuery({
+    queryKey: ['vaultDetails'],
+    queryFn: async () => {
+      return wrapNullable(
+        VaultCalculator.calculateVaultSummary(
+          VaultCalculator.getVaultHistoricalPnlResponse(await placeholderFetchMegavaultHistory())
+        )
+      );
+    },
+    ...vaultQueryOptions,
+  });
+
+  return vaultDetails?.data;
```

Comment:
why not return entire query but overwrite data?

```
return {
  ...vaultDetailsQuery,
  data: vaultDetailsQuery.data?.data
}
```

Reply:
As to not lose access to other query info

Reply:
ehh I don't love returning all that information but i do think having at least `status` will be helpful so I did it. Note that for hooks and combine multiple useQuery results we kind of don't have an object to return so just return the result, which is back where we started (and inconsistent with the others now)